android/src/main/java/com/polidea/flutter_ble_lib/FlutterBleLibPlugin.java

```diff
@@ -64,6 +76,12 @@ public void onMethodCall(MethodCall call, Result result) {
             case MethodName.STOP_DEVICE_SCAN:
                 stopDeviceScan(result);
                 break;
+            case MethodName.CONNECT_TO_DEVICE:
+            case MethodName.IS_DEVICE_CONNECTED:
+            case MethodName.OBSERVE_CONNECTION_STATE:
+            case MethodName.CANCEL_CONNECTION:
+                deviceConnectionDelegate.onMethodCall(call, result);
```

Comment:
why does the connection have separate delegate but scan does not?

Reply:
We kind of forgot about it. I can move scanning to a separate delegate in next PR.

Reply:
👍

Reply:
btw I would like to refactor that :) it is nice entry task :)

Reply:
Created an issue for this.